### PR TITLE
Base scraper

### PR DESCRIPTION
**Pull request by ennumramdan**
Jira link: https://juloprojects.atlassian.net/browse/NEW-35

### Purpose:
This code change is for build base scraper engine

#### Added:
* base scrape abrstract class
* bca scraper (done)
* mandiri scraper (login & logout)
* bri scraper (login & logout)

* base browser Firefox & Chrome (using context manager)

Based on [JULO Code Review Guide](https://docs.google.com/document/d/1Mh-NwB_BQXH7xfoDs-oKwGrALWV93l9seIyE36OkFiE/edit?usp=sharing).

### CODE REVIEW

File: bank_scraper/bri.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+from bank_scraper.browser import FirefoxBrowser, ChromeBrowser
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import Select, WebDriverWait
+from selenium.common.exceptions import NoAlertPresentException
+from selenium.common.exceptions import NoSuchElementException
+from selenium.common.exceptions import TimeoutException
+from selenium.webdriver.support import expected_conditions as EC
+
+class BriScraper(BaseScraper):
```
Comment (rikki-yanhari) [inline bank_scraper/bri.py]: I think this init method is quite similar in a couple first line, it's better to define it in abstract init method the you can add the the rest line in child class init method

File: bank_scraper/bca.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+        while time.time() < t_end:
+            if len(glob.glob1(self.raw_dir, "*.pdf")) >= n:
+                break
+
+            time.sleep(5)
+
+        if len(glob.glob1(self.raw_dir, "*.pdf")) < n:
+            print("some pdf failed to download")
+
+    def get_captcha_text(self):
```
Comment (rikki-yanhari) [inline bank_scraper/bca.py]: no need to defined if you don't use it

File: bank_scraper/bri.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        right = location['x'] + size['width']
+        bottom = location['y'] + size['height']
+
+        im = im.crop((left, top, right, bottom)) # defines crop points
+        im = im.convert('L') # grayscale
+        custom_config = r'--oem 3 --psm 6 -c tessedit_char_whitelist=0123456789'
+
+        return image_to_string(im, config=custom_config)
+
+    def get_account_info(self):
```
Comment (rikki-yanhari) [inline bank_scraper/bri.py]: also this method is not necessary

File: bank_scraper/browser.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        print([exc_type, exc_value, traceback])
+
+        self.browser.quit()
+        if self.virtual_display:
+            self.xvfb.stop()
+
```
Comment (rikki-yanhari) [inline bank_scraper/browser.py]: too many blank line

File: bank_scraper/mandiri.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    def get_account_info(self):
+        pass
+
+    def get_transaction_history(self):
+        pass
+
+    def downloan_e_statement(self):
+        pass
+
+    def get_captcha_text(self):
```
Comment (rikki-yanhari) [inline bank_scraper/mandiri.py]: no need to define if you not using that method

File: bank_scraper/bri.py (hunk 3 above)
Comment (ennumramdan) [reply bank_scraper/bri.py]: not finish yet

File: bank_scraper/bca.py (hunk 2 above)
Comment (ennumramdan) [reply bank_scraper/bca.py]: get error this if i not declare the method
```
TypeError: Can't instantiate abstract class BcaScraper with abstract methods get_captcha_text
```

**Pull request merged**